src/db.rs

```diff
@@ -45,6 +45,8 @@ impl DbInner {
             memtable_flush_notifier,
         };
         db_inner.replay_wal().await?;
+        db_inner.state.write().add_empty_wal();
```

Comment:
shall we condition this on `db_inner.wal_enabled()` ? If the wal is not enabled we don't really need to do this fencing.

Reply:
also, can we move these into some method like `fence_writers`

Reply:
I wonder if it makes sense to write these WAL entries from a codepath separate from the core flush loop.

My concern is how we'll implement the full protocol where we need to take a slightly different path for these WAL entries that looks like:
1. try to write a fencing WAL entry W with the new epoch E
2. if the write fails due to an "already exist error", check the WAL entry epoch. If it's lower than E, then increment the entry to W + 1 and try again.

Whereas the logic for flushing actual WAL entries should look like:
1. try to write the WAL entry. On a conflict, consider the DB fenced (the conflicting entry has a higher epoch) or panic (the conflicting entry has the same or lower epoch).

I think it will be easier to get this right if the core flush loop just handles the second case and have the fencing logic be separate. It would make me nervous to have the core flush logic try to retry writing WAL entries when its fenced.

Reply:
Above approach to special case fencing flush  and keeping the current flush only for data flush does look better.
Current approach would write a manifest with, fail with fenced error and exit. Old writer would continue writing until it reads the new manifest after manifest poll interval, and will exit at that point.

I will change this.
I am thinking of getting the wal list, apply the logic shared above and then pass the wal list to `replay_wal` for replay, this would also fence the writers before replay is done.

Reply:
I am fencing first, and then replaying. It also handles the case where there are more wals after a manifest update.